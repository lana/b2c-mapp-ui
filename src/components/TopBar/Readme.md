# TopBar
A custom native AppBar replacement. It only renders the title, all actions should be placed by the Android native bridge.

| Prop |  Value Type | Required |
| --- | --- | --- |
| `className` | `String` | false | 
| `title` | `String` | false | 

```
<TopBar title='My custom title' />
```
