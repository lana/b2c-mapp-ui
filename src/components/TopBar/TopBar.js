import Heading from '../Heading/Heading.vue';

const components = {
  Heading,
};

const props = {
  dataTestId: {
    type: String,
    default: 'topbar-header',
  },
  title: {
    type: String,
    default: '',
  },
};

const TopBar = {
  components,
  props,
};

export default TopBar;
