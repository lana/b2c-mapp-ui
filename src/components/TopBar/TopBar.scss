@import '../../styles/colors';

.topbar {
  position: relative;
  z-index: 95;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  flex-shrink: 0;
  height: 56px;
  padding: 0 56px 0 72px;
  text-align: left;
  background: $white;
}

.title {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}
