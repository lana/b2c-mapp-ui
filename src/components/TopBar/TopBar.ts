import { defineComponent } from 'vue';

import Heading from '../Heading/Heading.vue';

const TopBar = defineComponent({
  name: 'TopBar',
  components: {
    Heading,
  },
  props: {
    dataTestId: {
      type: String,
      default: 'topbar-header',
    },
    title: {
      type: String,
      default: '',
    },
  },
});

export default TopBar;
