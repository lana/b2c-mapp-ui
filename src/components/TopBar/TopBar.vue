<template>
  <header :data-testid="dataTestId" class="topbar">
    <Heading class="headline title">
      {{ title }}
    </Heading>
  </header>
</template>

<script lang="ts" src="./TopBar.ts"></script>
<style lang="scss" scoped src="./TopBar.scss"></style>
