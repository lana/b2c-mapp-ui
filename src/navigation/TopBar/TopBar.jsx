import { useMemo } from 'preact/hooks';
import PropTypes from 'prop-types';
import CSS from './styles.css';
import Heading from '../../typography/Heading/Heading';

const TopBar = ({ className, title, dataTestId }) =>
	useMemo(
		() => (
			<header data-testid={dataTestId} className={`${CSS.topbar} ${className}`}>
				<Heading type={`headline`} className={CSS.title}>
					{title}
				</Heading>
			</header>
		),
		[title],
	);

TopBar.defaultProps = {
	className: '',
	dataTestId: 'topbar-header',
};

TopBar.propTypes = {
	className: PropTypes.string,
	title: PropTypes.string.isRequired,
	dataTestId: PropTypes.string,
};

export default TopBar;