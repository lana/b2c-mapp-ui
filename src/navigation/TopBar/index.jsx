import CSS from './styles.css'
import Heading from '../../typography/Heading'

export default ({ className, title, dataTestId }) => (
	<header data-testid={dataTestId || 'topbar-header'} className={`${CSS.topbar} ${className}`}>
		<Heading type={`headline`} className={CSS.title}>
			{title}
		</Heading>
	</header>
)
