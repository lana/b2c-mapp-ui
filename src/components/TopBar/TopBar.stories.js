import TopBar from './TopBar.vue';
import { createDeviceDecorator } from '../../lib/storybookHelpers';

const deviceDecorator = createDeviceDecorator('<strong>TopBar:</strong>&nbsp;A custom native AppBar replacement.', '<p style="margin-top: 10px;">It only renders the title, all actions should be placed by the Android native bridge.</p>');

const TopBarStories = {
  component: TopBar,
  title: 'Components/TopBar',
  decorators: [deviceDecorator],
  args: {
    ...deviceDecorator.args,
    title: 'Example custom title',
  },
  argTypes: {
    ...deviceDecorator.argTypes,
    title: { control: 'text', name: 'Title' },
  },
};

const defaultExample = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  setup() { return { ...args }; },
  components: {
    TopBar,
  },
  template: '<TopBar :title="title"/>',
});
defaultExample.parameters = {
  docs: {
    source: {
      code: '<TopBar :title="title"/>',
    },
  },
};

export {
  defaultExample,
};

export default TopBarStories;
