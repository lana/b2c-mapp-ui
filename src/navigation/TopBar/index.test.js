import { render } from '@testing-library/preact';
import TopBar from './index';

describe('UI/navigation/TopBar', () => {

    const defaultProps = {
        className: 'test-classname',
        dataTestId:'top-bar',
        title: 'my title'
    }

    it('Should show given title', () => {
        const { getByTestId } = render(<TopBar {...defaultProps}/>)
        const title = getByTestId('heading');
        expect(title.textContent).toEqual('my title');
    })

    it('Should apply given classname to header', () => {
        const { getByTestId } = render(<TopBar {...defaultProps}/>)
        const header = getByTestId('top-bar');
        expect(header.className.includes('test-classname')).toBe(true);
    })

});