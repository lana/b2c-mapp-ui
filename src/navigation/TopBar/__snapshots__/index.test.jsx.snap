// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/navigation/TopBar should render correctly 1`] = `ReactWrapper {}`;
